File: App/frontend/app-movil/lib/services/servidor.dart
import 'dart:io' show Platform;

/// Dónde está el backend.
///
/// Una sola dirección: el **API Gateway** (ADR-02), que autentica cada petición
/// y la enruta al microservicio correspondiente. La app no sabe cuántos
/// servicios hay ni en qué puerto está cada uno.
///
/// | Ruta | Va a |
/// |---|---|
/// | `/api/v1/sesiones`, `/api/v1/cuentas` | Administración (CU-027) |
/// | `/api/v1/reventa` | Entradas y Mercado Secundario (CU-006) |
///
/// El emulador de Android no ve el `localhost` de la máquina anfitriona: para
/// él `localhost` es el propio emulador. `10.0.2.2` es el alias que Android
/// reserva justo para esto. El simulador de iOS sí comparte red con el Mac.
///
/// En un teléfono físico hay que indicar la IP del computador:
///
///   flutter run --dart-define=HEXACORE_HOST=192.168.18.16
///
/// `HEXACORE_API` apunta a un gateway completo distinto (por ejemplo, uno
/// desplegado).
class Servidor {
  Servidor._();

  static const _host = String.fromEnvironment('HEXACORE_HOST');
  static const _api = String.fromEnvironment('HEXACORE_API');

  static String _hostPorDefecto() {
    if (_host.isNotEmpty) return _host;
    if (_api.isNotEmpty) return Uri.parse(_api).host;
    return Platform.isAndroid ? '10.0.2.2' : 'localhost';
  }

  /// API Gateway: el único punto de entrada al backend.
  static String get api => _api.isNotEmpty ? _api : 'http://${_hostPorDefecto()}:8080';

  /// Buzón del proveedor de correo simulado. **Solo tiene sentido en
  /// desarrollo**: ahí llegan los enlaces de verificación y recuperación, y no
  /// pasa por el gateway porque es un sistema externo simulado.
  static String get buzonDesarrollo => 'http://${_hostPorDefecto()}:3098/correos';

  static const prefijo = 'api/v1';
}

File: App/frontend/portal-web-cliente/src/app/core/servidor.ts
/**
 * Dónde está el backend.
 *
 * Una sola dirección: el **API Gateway** (ADR-02), que autentica y enruta hacia
 * el microservicio que corresponda. Antes había que conocer el puerto de cada
 * servicio; ahora el portal no sabe cuántos hay ni dónde están.
 *
 * | Ruta | Va a |
 * |---|---|
 * | `/api/v1/sesiones`, `/api/v1/cuentas`, `/api/v1/admin` | Administración (CU-027) |
 * | `/api/v1/reventa` | Entradas y Mercado Secundario (CU-006) |
 *
 * Se usa el mismo nombre de máquina con que se abrió el portal, así funciona
 * igual en `localhost` que abriéndolo desde otro equipo de la red.
 */
const host = typeof window === 'undefined' ? 'localhost' : window.location.hostname;

export const Servidor = {
  /** API Gateway: el único punto de entrada al backend. */
  api: `http://${host}:8080/api/v1`,

  /**
   * Buzón del correo simulado. **Solo desarrollo**, y no pasa por el gateway:
   * no es parte del sistema, sino el proveedor de notificaciones simulado
   * (`App/infra/correo-simulado`), donde se leen los enlaces de los correos.
   */
  buzonDesarrollo: `http://${host}:3098/correos`
} as const;
